### ISSUE TITLE
Logs (log4net) folder does not respect the -datadir when working folder different from .dll

### ISSUE THREAD

**Issue by sondreb** [stratisproject/StratisBitcoinFullNode#1307]
When launching the daemon from the .csproj location using PowerShell and specifying the working directory, the wallet files (written with FileStorage) is written relative to the working folder (which I would expect is correct), while the Logs folder is written relative to the .dll files, which is wrong.

```
Start-Process -FilePath 'dotnet' -WorkingDirectory 'C:\src\Stratis.StratisD' -ArgumentList 'run -testnet -datadir="StratStore1"'
```

I have fixed the issue and will supply a PR related to this bug report. The tests supplied are to verify that the implementation code is valid, but the tests doesn't test the actual code due to not very testable. I considered refactored code to make it test-able, but decided not to to make the PR smaller.

**Comment by Aprogiena** [comment]
would it not make better sense to ensure that LogPath (and other paths in DataFolder) are rooted?

**Comment by Aprogiena** [comment]
what i mean is that it makes better sense to me to address the issue in DataFolder rather then in each end code when it is used

**Comment by sondreb** [comment]
Could be an idea to root all the paths in DataFolder, but that would be a bigger change that would need additional integration testing to verify. Perhaps accept this PR, and then we could investigate if rooting all the paths would make sense?

**Comment by Aprogiena** [comment]
If there is issue with one path, it can very much happen with any other path. Fixing it at 1 place for everything seems to be better approach than just to fix every particular instance one by one. So I would suggest to go that way instead of this PR, but feel free to wait for opinions of others.

**Comment by sondreb** [comment]
The issue is really with path for log files, produced by log4net, currently can end up somewhere other than the normal data files.

Changing DataFolder to root all the paths is likely a major refactoring job, considering there are hundreds of usage of DataFolder and many File IO operations done throughout the code base. If even a single one of those read/write operations changes position, it can be very problematic.

This does though bring up a potential new PR, and that is to rewrite all code that uses System.IO directly and make sure every interaction with the file system goes through the FileStorage class, as of right now it's spread a bit throughout the code base. Then when all IO is handled through the FileStorage abstraction, it is possible to be more certain that there won't be an issues. I wouldn't start rooting the paths unless that was done first.

**Comment by Aprogiena** [comment]
That has to be some kind of misunderstanding. My idea was only to ensure that the DataFolder class does not have any relative paths. This can be achieved by a trivial change in a single function. Then it should not matter which API is used for working with those paths because they will be rooted. So in terms of complexity this would be a slight PR as your original, except that it would fix all potential problems with all paths that are stored by data folder class. Or am I missing something why would that not work?

**Comment by dangershony** [comment]
I would very much like that the NLog.config will be picked from datafolder can we make this happen?

**Comment by Aprogiena** [comment]
@dangershony This does not seem to be relevant to the issue. But anyway, what you say is possible, but prevents logging during early initialisation of the node. This is why we decided initially not to put it into the data folder. It would only make sense to do that if we separated loading of configuration file, such that it happens is the 1st thing in the node, before all other initialisation starts. I'm not sure if we can do that. But anyway, it does not seem to be related to this issue, that is different discussion I believe.

**Comment by dangershony** [comment]
Yeah I am aware but as a side note I think putting log configuration in data folder is very useful I think we should try to make it happen

**Comment by dangershony** [comment]
It is from a user experience most users know where the data folder is and its easy to just drop a file there (our other config files are there already) ok I will create an issue.

**Comment by bokobza** [comment]
@sondreb 
> the wallet files (written with FileStorage) is written relative to the working folder (which I would expect is correct), while the Logs folder is written relative to the .dll files, which is wrong.

where do you see that the logs folder is written relative to the dlls? I can't see it 🧐
